docs/protocol/specifications/blockchain.md

```diff
@@ -0,0 +1,613 @@
+
+
+#### Output Validation
+
+1. [Validate](#value-source-1-validation) `Source`.
+2. If the transaction version is 1: verify that the `ExtHash` is the all-zero hash.
+
+
+#### Retirement 1
+
+Field               | Type                 | Description
+--------------------|----------------------|----------------
+Type                | String               | "retirement1"
+Body                | Struct               | See below.
+Witness             | Struct               | Empty struct.
+
+Body field          | Type                 | Description
+--------------------|----------------------|----------------
+Source              | ValueSource1         | The source of the units that are being retired.
+Data                | Hash                 | Hash of the reference data for this entry, or a string of 32 zero-bytes (representing no reference data).
+ExtHash             | [ExtStruct](#extension-struct) | If the transaction version is known, this must be 32 zero-bytes.
+
+#### Retirement Validation
+
+1. [Validate](#value-source-1-validation) `Source`.
+2. If the transaction version is 1: verify that the `ExtHash` is the all-zero hash.
+
+### Spend 1
+
+Field               | Type                 | Description
+--------------------|----------------------|----------------
+Type                | String               | "spend1"
+Body                | Struct               | See below.
+Witness             | Struct               | See below.
+
+Body field          | Type                 | Description
+--------------------|----------------------|----------------
+SpentOutput         | Pointer<Output1>      | The Output entry consumed by this spend.
+Data                | Hash                 | Hash of the reference data for this entry, or a string of 32 zero-bytes (representing no reference data).
+ExtHash             | [ExtStruct](#extension-struct) | If the transaction version is known, this must be 32 zero-bytes.
+
+Witness field       | Type                 | Description
+--------------------|----------------------|----------------
+Destination         | ValueDestination1    | The Destination ("forward pointer") for the value contained in this spend. This can point directly to an Output entry, or to a Mux, which points to Output entries via its own Destinations.
+Arguments           | List<String>         | Arguments for the control program contained in the SpentOutput.
+AnchoredEntry       | Pointer                      | Optional pointer to a single entry of any type, which uniquely identifies that entry as one that can use this one as an `Anchor`.
+
+#### Spend Validation
+
+1. Verify that `SpentOutput` is present in the transaction, but do not validate it.
+2. [Validate program](#program-validation) `SpentOutput.ControlProgram` with the given `Arguments` and the transaction version.
+3. Verify that `SpentOutput.Value` is equal to `Destination.Value`.
+4. [Validate](#value-destination-1-validation) `Destination`.
```

Comment:
Hm, I take it back - validation of `ValueDestination`s does not recur into the referenced entry, so this is safe.

However, it may be worth adding some language emphasizing this so casual readers don't think there's an infinite regress here.